spring-client-common/src/main/java/io/camunda/commons/auth/SelfManagedAuthentication.java

```diff
@@ -0,0 +1,115 @@
+package io.camunda.commons.auth;
+
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.lang.invoke.MethodHandles;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static io.camunda.commons.http.JsonUtils.toResult;
+
+public class SelfManagedAuthentication extends JwtAuthentication {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private String authUrl;
+  private String keycloakRealm = "camunda-platform";
+  private String keycloakUrl;
+  private JwtConfig jwtConfig;
+  private Map<Product, String> tokens;
+  private Map<Product, Integer> expirations;
+
+  public SelfManagedAuthentication() {
+    tokens = new HashMap<>();
+    expirations = new HashMap<>();
+  }
+
+  public SelfManagedAuthentication jwtConfig(JwtConfig jwtConfig) {
```

Comment:
I am not sure how we align on this from DevEx perspective, but I have not seen a builder on the actual class in a Camunda 8 API before..

I would separate the builder from the class and offer a constructor for all args.

This allows for properly setting lifecycle binding of fields (by making them final).

Reply:
Separated the builder from the class in https://github.com/camunda-community-hub/spring-zeebe/pull/510/commits/603bb6a40b1d5295d7d805c0648569b90972389c

However I don't get where do you suggest to put the all args constructor. I guess that goes in the class itself? So we would offer users two ways to use a class: either thru the builder or all args. 

I don't quite get the `property setting lifecylcle bindings of fields`. What advantage or use would that give us? Any examples maybe from Zeebe?